test-suite/unit-tests/tools/utest_rules.ml

```diff
@@ -0,0 +1,44 @@
+open Format
+
+let subdirs =
+  [ "lib", "coq.lib"
+  ; "clib", "coq.clib"
+  ; "printing", "coq.printing"
+  ]
+
+let filter_ext ~ext files =
+  List.filter (fun file -> Filename.check_suffix file ext) files
+
+let replace_ext ~ext file =
+  Filename.remove_extension file ^ ext
+
+let gen_prelude ~libraries fmt files =
+  let exes = List.map Filename.remove_extension files in
+  fprintf fmt
+    "(executables@\n (names %a)@\n (libraries coq_utest %s))"
+    (pp_print_list pp_print_string) exes libraries
+
+let output_log_rule fmt file =
+  let exe = replace_ext ~ext:".exe" file in
+  fprintf fmt
+    "(rule@\n (targets %s.log)@\n (action (with-accepted-exit-codes 0 (run ./%s))))"
```

Comment:
Isn't accepting only the `0` code the default behavior? Why do you need to add this?

Reply:
Yes sorry, I added just to be sure that the return value was taken into account, it turns out that our ounit implementation is buggy and always does `exit 0` ....